### DISCUSSION TITLE
Open profile page with PubKey

### DISCUSSION THREAD

**Discussion by tijno** [deso-protocol/dips#80] [Medium CIPs]
**Suggested change:**

Allow use of PublicKey in user profile URLs, eg: `https://bitclout.com/u/BC1YLgxLrxvq5mgZUUhJc1gkG6pwrRCTbdT6snwcrsEampjqnSD1vck` which then looks up the username and shows `https://bitclout.com/u/tijn`

**Rationale** 

Allow linking to user profiles that frequently change their username. Also prepare road for using PubKey in mentions.

**Implementation**

Add a check here 

https://github.com/bitclout/frontend/blob/main/src/app/creator-profile-page/creator-profile-details/creator-profile-details.component.ts#L49

To determine if the path is a pub key:

```
params.username.substring(0, 2) != "BC" && params.username.length < 50
```

If needed - lookup the username with [GetSingleProfile api request](https://github.com/bitclout/backend/blob/main/routes/user.go#L1015)

Then assign username to `this.userName`

It may be possible to do this without a Username lookup api call here:

https://github.com/bitclout/frontend/blob/e17cc029e24b15ca36d3b41daf2a7a0275dcc55a/src/app/creator-profile-page/creator-profile-details/creator-profile-details.component.ts#L167

Either way - after loading user data must update route to the `/u/username` version.

**Future Enhancement**

Ones [@profile mentions](https://github.com/bitclout/cips/discussions/35) have been implemented, a mention could actually encode the pub key in the post, eg `@BC1YLgxLrxvq5mgZUUhJc1gkG6pwrRCTbdT6snwcrsEampjqnSD1vck` so that when clicked it opens the profile page of the mentioned account.

**Comment by tijno** [comment]
Any questions or suggestions on this medium proposal?

Its got 3 thumbs up - so looks like community is ok with it.

**Reply by ItsAditya-xyz** [reply]
Definitely a positive change Tijn!
I wonder what would happen if you search for shadow accounts 🤔

**Reply by tijno** [reply]
@AdityaChaudhary0005 Same as when searching for a username - show 404

**Comment by sungkhum** [comment]
Yes please

**Comment by brootle** [comment]
I approve ✅
